src/main/java/net/sf/jabref/shared/DBMSProcessor.java

```diff
@@ -0,0 +1,562 @@
+            }
+        } catch (SQLException e) {
+            LOGGER.error("SQL Error", e);
+        }
+
+        return sharedIDVersionMapping;
+    }
+
+    /**
+     * Fetches and returns all shared meta data.
+     */
+    public Map<String, String> getSharedMetaData() {
+        Map<String, String> data = new HashMap<>();
+
+        try (ResultSet resultSet = dbmsHelper.query("SELECT * FROM " + escape("METADATA"))) {
+            while(resultSet.next()) {
+                data.put(resultSet.getString("KEY"), resultSet.getString("VALUE"));
+            }
+        } catch (SQLException e) {
+            LOGGER.error("SQL Error", e);
+        }
+
+        return data;
+    }
+
+    /**
+     * Clears and sets all shared meta data.
+     * @param metaData JabRef meta data.
+     */
+    public void setSharedMetaData(Map<String, String> data) {
+        dbmsHelper.executeUpdate("TRUNCATE TABLE " + escape("METADATA")); // delete data all data from table
+
+        for (Map.Entry<String, String> metaEntry : data.entrySet()) {
+
+            StringBuilder query = new StringBuilder()
+            .append("INSERT INTO ")
+            .append(escape("METADATA"))
+            .append("(")
+            .append(escape("KEY"))
+            .append(", ")
+            .append(escape("VALUE"))
+            .append(") VALUES(?, ?)");
+
+            try (PreparedStatement preparedStatement = dbmsHelper.prepareStatement(query.toString())) {
+                preparedStatement.setString(1, metaEntry.getKey());
+                preparedStatement.setString(2, metaEntry.getValue());
+                preparedStatement.executeUpdate();
+            } catch (SQLException e) {
+                LOGGER.error("SQL Error: ", e);
+            }
+        }
+    }
+
+    /**
+     * Helping method for SQL selection retrieving a {@link ResultSet}
+     * @param sharedID Existent ID of {@link BibEntry} on shared database
+     */
+    private ResultSet selectFromEntryTable(int sharedID) throws SQLException {
+        StringBuilder query = new StringBuilder()
+        .append("SELECT * FROM ")
+        .append(escape("ENTRY"))
+        .append(" WHERE ")
+        .append(escape("SHARED_ID"))
+        .append(" = ")
+        .append(sharedID);
+
+        return dbmsHelper.query(query.toString());
+    }
+
+    /**
+     * Escapes parts of SQL expressions like table or field name to match the conventions
+     * of the database system.
+     * @param expression Table or field name
+     * @param type Type of database system
+     * @return Correctly escape expression
+     */
+    public static String escape(String expression, DBMSType type) {
+        if (type == DBMSType.MYSQL) {
+            return "`" + expression + "`";
+        } else if ((type == DBMSType.ORACLE) || (type == DBMSType.POSTGRESQL)) {
+            return "\"" + expression + "\"";
+        }
+        return expression;
```

Comment:
This escape method should be an instance method of the DBMSType, as it only depends on this.